templates/vec.rs.tera

```diff
@@ -4080,39 +4080,7 @@ impl From<{{ macros::make_tuple_t(t=scalar_t, n=dim) }}> for {{ self_t }} {
 impl From<{{ self_t }}> for {{ macros::make_tuple_t(t=scalar_t, n=dim) }} {
     #[inline]
     fn from(v: {{ self_t }}) -> Self {
-        {% if is_scalar %}
-            (
```

Comment:
It turns out that in an optimized build just returning `(self.x, self.y, ...)` results in the same code as the other two methods. https://rust.godbolt.org/z/5rWvsE3T7

So it would be better to just change this to

```
            (
                {% for c in components %}
                    v.{{ c }},
                {%- endfor %}
            )
```